### ISSUE TITLE
Add `upgrade-insecure-requests` directive to Content Security Policy

### ISSUE THREAD

**Issue by robhudson** [mozilla/bedrock#15556]

**Comment by janbrasna** [comment]
This seems to break local development in Safari (and Webkit-based Firefox iOS and Focus iOS) that upgrade even loopback requests:

<img width="999" alt="Screenshot 2024-12-22 at 18 45 03" src="https://github.com/user-attachments/assets/5fdb742d-7043-4461-b7f7-050d99fcbbca" />

Different implementations trust localhost (e.g. for mixed content) differently; the same is carried over to `upgrade-insecure-requests`:
- Gecko works: https://bugzilla.mozilla.org/show_bug.cgi?id=1447784
- WebKit breaks: https://bugs.webkit.org/show_bug.cgi?id=250776

Basically only requests already server over HTTPS should ask to upgrade anything that follows from HTTP. (Or, alternatively, find a stable way to skip that rule when used locally.)

**Comment by robhudson** [comment]
Interesting. We could consider if `DEBUG` is True, make this `False`. That should cover the typical local development cases but not include demos and production environments (dev/stage/prod), etc.

**Comment by janbrasna** [comment]
Given the recent "localmess"-gate and not much progress on Apple side to validate various hosts mapped to loopback to provide WebKit with the groundwork it needs to make decisions for mixed content, this looks like something we need to live with for the foreseeable future. Seems to work well.
